user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleBatchInformationSpreadsheet.java

```diff
@@ -0,0 +1,384 @@
+package life.qbic.datamanager.views.projects.project.samples.registration.batch;
+
+import static java.util.Objects.isNull;
+import static java.util.function.Function.identity;
+
+import com.vaadin.flow.component.html.Span;
+import com.vaadin.flow.component.icon.VaadinIcon;
+import com.vaadin.flow.data.renderer.ComponentRenderer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.StringJoiner;
+import java.util.stream.Collectors;
+import life.qbic.datamanager.views.general.spreadsheet.Column;
+import life.qbic.datamanager.views.general.spreadsheet.Spreadsheet;
+import life.qbic.datamanager.views.projects.project.samples.registration.batch.SampleBatchInformationSpreadsheet.SampleInfo;
+import life.qbic.projectmanagement.domain.model.experiment.BiologicalReplicate;
+import life.qbic.projectmanagement.domain.model.experiment.Condition;
+import life.qbic.projectmanagement.domain.model.experiment.ExperimentalGroup;
+import life.qbic.projectmanagement.domain.model.experiment.vocabulary.Analyte;
+import life.qbic.projectmanagement.domain.model.experiment.vocabulary.Species;
+import life.qbic.projectmanagement.domain.model.experiment.vocabulary.Specimen;
+import life.qbic.projectmanagement.domain.model.sample.AnalysisMethod;
+import life.qbic.projectmanagement.domain.model.sample.SampleId;
+
+/**
+ * A spreadsheet used for sample batch information
+ */
+public class SampleBatchInformationSpreadsheet extends Spreadsheet<SampleInfo> {
+
+
+  private final List<ExperimentalGroup> experimentalGroups;
+  private final List<Species> species;
+  private final List<Specimen> specimens;
+  private final List<Analyte> analytes;
+
+  private static final int INITIAL_ROW_COUNT = 2;
+  private final Column<SampleInfo, String> sampleCodeColumn;
+
+
+  public SampleBatchInformationSpreadsheet(List<ExperimentalGroup> experimentalGroups,
+      List<Species> species, List<Specimen> specimens, List<Analyte> analytes) {
+    this.experimentalGroups = new ArrayList<>(experimentalGroups);
+    this.species = new ArrayList<>(species);
+    this.specimens = new ArrayList<>(specimens);
+    this.analytes = new ArrayList<>(analytes);
+    List<AnalysisMethod> sortedAnalysisMethods = Arrays.stream(AnalysisMethod.values())
+        .sorted(Comparator.comparing(AnalysisMethod::label))
+        .toList();
+
+    sampleCodeColumn = addColumn("Sample code",
+        sampleInfo -> sampleInfo.getSampleCode().orElse(""),
+        SampleInfo::setSampleCode)
+        .requireDistinctValues();
+    lockColumn(sampleCodeColumn);
+    hideColumn(sampleCodeColumn);
```

Comment:
The column was not hidden for me which is confusing if the user wants to create a batch since no samplecodes can be provided there
![Bildschirmfoto 2023-11-23 um 13 57 55](https://github.com/qbicsoftware/data-manager-app/assets/29627977/8e88425d-8eb1-4cc4-a71d-ba0d0288f478)